compiler-core/src/build/package_loader.rs

```diff
@@ -162,9 +166,18 @@ where
         };
 
         // Src
-        for path in self.io.gleam_source_files(&src) {
-            let input = loader.load(path)?;
-            inputs.insert(input)?;
+        for path in self.io.gleam_ext_files(&src) {
+            if self.io.gleam_source_files(&src).contains(&path) {
```

Comment:
This iterates over the Gleam files on disc count^2 times. We can instead do something like this:

```rust
for path in self.io.gleam_files(&src) {
    if !valid_module_path(path) {
        emit_warning();
        continue;
    }
    // ...
```